Samples-Typescript/DashboardLayout/dashboardLayout.ts

```diff
@@ -0,0 +1,155 @@
+          toAppend.text(`Dashboard Object ${dashboardObjectId} removed: "${oldDashboardObject.name}"`);
+          changesDiv.append(toAppend);
+          this._$('#dashboard-layout-change-list').append(changesDiv);
+          return;
+        }
+
+        // the following dashboard changes are not mutually exclusive, so we list them together.
+        const h6 = this._$('<h6/>');
+        h6.text(`Dashboard Object ${dashboardObjectId}: "${dashboardObject.name}"`);
+        changesDiv.append(h6);
+        const ul = this._$('<ul/>');
+
+        // checking if the dashboard object had changes to its floating state.
+        if (changesMade.includes(tableau.DashboardLayoutChange.IsFloatingChanged)) {
+          const li = this._$('<li/>');
+          li.text(`Floating is now ${dashboardObject.isFloating}, was ${oldDashboardObject.isFloating}`);
+          ul.append(li);
+        }
+        // checking if the dashbaord object had changes to its visibility.
+        if (changesMade.includes(tableau.DashboardLayoutChange.IsVisibleChanged)) {
+          const li = this._$('<li/>');
+          li.text(`Visibility is now ${dashboardObject.isVisible}, was ${oldDashboardObject.isVisible}`);
+          ul.append(li);
+        }
+        // checking if the dashboard object was repositioned.
+        if (changesMade.includes(tableau.DashboardLayoutChange.PositionChanged)) {
+          const li = this._$('<li/>');
+          const newPos = dashboardObject.position;
+          const oldPos = oldDashboardObject.position;
+          li.text(`Position is now (${newPos.x},${newPos.y}), was (${oldPos.x},${oldPos.y})`);
+          ul.append(li);
+        }
+        // checking if the dashboard object was resized.
+        if (changesMade.includes(tableau.DashboardLayoutChange.SizeChanged)) {
+          const li = this._$('<li/>');
+          const newSize = dashboardObject.size;
+          const oldSize = oldDashboardObject.size;
+          li.text(`Size is now ${newSize.width}x${newSize.height}, was ${oldSize.width}x${oldSize.height}`);
+          ul.append(li);
+        }
+        // checking if the dashboard object was renamed.
+        if (changesMade.includes(tableau.DashboardLayoutChange.NameChanged)) {
+          const li = this._$('<li/>');
+          li.text(`Name is now "${dashboardObject.name}", was "${oldDashboardObject.name}"`);
+          ul.append(li);
+        }
+
+        changesDiv.append(ul);
+        this._$('#dashboard-layout-change-list').append(changesDiv);
+      });
+    }
+
+    // This function adds a dashboard event if there is not one already, and removes it if there is.
+    private onEventButtonClick() {
+      const dashboard = tableau.extensions.dashboardContent.dashboard;
+      if ($('#dashboard-event-btn').text() === 'Add Dashboard Event') {
+        // tslint:disable-next-line: max-line-length
+        dashboard.addEventListener(tableau.TableauEventType.DashboardLayoutChanged, (event) => this.onDashboardLayoutChange(event));
```

Comment:
dashboard.addEventListener(
  tableau.TableauEventType.DashboardLayoutChanged, 
  (event) => this.onDashboardLayoutChange(event));